crates/iceberg/src/catalog/mod.rs

```diff
@@ -446,6 +446,30 @@ pub enum TableUpdate {
         /// Properties to remove
         removals: Vec<String>,
     },
+    /// Set statistics for a snapshot
+    #[serde(with = "_serde_set_statistics")]
+    SetStatistics {
+        /// File containing the statistics
+        statistics: StatisticsFile,
+    },
```

Comment:
In IRC and Java `SetStatistics` has an additional field `snapshot_id` ([link](https://github.com/apache/iceberg/blob/540d6a6251e31b232fe6ed2413680621454d107a/open-api/rest-catalog-open-api.yaml#L2902)).
This field is redundant with `StatisticsFile.snapshot_id` and is only used as an [assertion in Java](https://github.com/apache/iceberg/blob/540d6a6251e31b232fe6ed2413680621454d107a/core/src/main/java/org/apache/iceberg/TableMetadata.java#L1314).

I removed the redundancy for rust and will start a discussion on the mailing List how to handle this.

As we still need to be spec compliant, we need custom serializer / deserializer.

Slack: https://apache-iceberg.slack.com/archives/C03LG1D563F/p1734109745807119

Reply:
Thanks for raising this on the dev-list: https://lists.apache.org/thread/6fnrp73wokfqlt5vormpjyjmtvl29ll1